### PR TITLE
feat(sdk-py): include experiment id in summary

### PR DESCRIPTION
**Pull request by sachinpad**

### CODE REVIEW

**Review by ankrgyl — APPROVED** [review]

File: py/src/braintrust/logger.py [hunk 1]
```diff
@@ -1942,6 +1942,7 @@ def summarize(self, summarize_scores=True, comparison_experiment_id=None):
 
         return ExperimentSummary(
             project_name=self.project.name,
+            experiment_id=self.id,
```
Comment (manugoyal) [inline py/src/braintrust/logger.py:1945]: fyi @ankrgyl: note an interesting consequence of this is that experiment summary with `summarize_scores=False` will now require a network roundtrip in order to resolve the id.

Comment (ankrgyl) [reply py/src/braintrust/logger.py:1945]: Oh interesting. Didn't it require one before to get the experiment's name?

Comment (manugoyal) [reply py/src/braintrust/logger.py:1945]: Ah yes good point. Never mind.

File: py/src/braintrust/logger.py [hunk 2]
```diff
@@ -2838,6 +2839,8 @@ class ExperimentSummary(SerializableDataClass):
 
     project_name: str
     """Name of the project that the experiment belongs to."""
+    experiment_id: Optional[str]
+    """ID of the experiment. May be None if the eval was run locally."""
```
[the remark this answers is not in the document]
Comment (sachinpad) [reply py/src/braintrust/logger.py]: Done

**Pull request merged**